src/test/groovy/com/hltech/pact/gen/domain/client/annotation/MappingHandlerFactorySpec.groovy

```diff
@@ -0,0 +1,52 @@
+package com.hltech.pact.gen.domain.client.annotation
+
+import com.hltech.pact.gen.domain.client.annotation.handlers.DeleteMappingMethodsHandler
+import com.hltech.pact.gen.domain.client.annotation.handlers.GetMappingMethodsHandler
+import com.hltech.pact.gen.domain.client.annotation.handlers.PatchMappingMethodsHandler
+import com.hltech.pact.gen.domain.client.annotation.handlers.PostMappingMethodsHandler
+import com.hltech.pact.gen.domain.client.annotation.handlers.PutMappingMethodsHandler
+import com.hltech.pact.gen.domain.client.annotation.handlers.RequestMappingMethodsHandler
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class MappingHandlerFactorySpec extends Specification {
+
+    private HandlersFactory handlersFactoryMock = Mock()
+
+    @Subject
+    private MappingHandlerFactory factory = new MappingHandlerFactory(handlersFactoryMock)
+
+    def 'should create all instances of handlers'() {
+        given:
+            final def expectedClasses = [
+                DeleteMappingMethodsHandler, GetMappingMethodsHandler, PatchMappingMethodsHandler,
+                PostMappingMethodsHandler, PutMappingMethodsHandler, RequestMappingMethodsHandler
+            ]
+
+            1 * handlersFactoryMock.createHandlers(_ as List<Class<?>>) >> { arguments ->
+                ((List<Class<?>>) arguments[0]).collect { it.newInstance() }
+            }
+
+        expect:
+            with(factory.createAll().collect { it.getClass() }) {
+                it.containsAll(expectedClasses)
```

Comment:
what if "it" contains "expectedClasses" and some unexpected? It will validate as true...

Reply:
I've added assertions for list size. This should do the job.